### ISSUE TITLE
implementing missing ff_eventcodes

### ISSUE THREAD

**Issue by ciscoyo** [gotzl/hid-fanatecff#47]
> General
Support for a bunch of effects, mostly copy-pasted from [new-lg4ff](https://github.com/berarma/new-lg4ff). Note: only tested with CSL Elite Wheelbase. Currently not supported effects: FF_FRICTION, FF_INERTIA

and i thought, its only me, coz i do not have the newest firmware..
ye, im stupid.. i should have just read the fukin descrition.. xD
(i deleted the other post about it ^^)

so again im offering my help/input/tests with this whenever! since its **the most important feedback** u want to have as a driver ^^ (and tbh, im still completely shocked that no one has ever even mentioned it before Oo). i mean, driving with sensing the current grip is way easier and faster than just trail and error and try memorise it (especially in f1 23, where u dont even have the audio input for it). than u r able to handle the over-understeer way better. and its creates a way more realistic immersion ofc ^^

i saw that @dtran11 did something similar here as well. [https://github.com/gotzl/hid-fanatecff/issues/45#issuecomment-1787556135](url)
it might help a bit. actually, ive heard that AI is really strong in these kind of thinks btw ^^

@gotzl i can close the other thread, no? im asking, coz i dont know the behavior here on github. and speaking about behavior, is this even the right tap or should i created it at pull requests?

**Comment by gotzl** [comment]
> so again im offering my help/input/tests with this whenever! since its the most important feedback u want to have as a driver ^^ (and tbh, im still completely shocked that no one has ever even mentioned it before Oo). i mean, driving with sensing the current grip is way easier and faster than just trail and error and try memorise it (especially in f1 23, where u dont even have the audio input for it). than u r able to handle the over-understeer way better. and its creates a way more realistic immersion ofc ^^
Why do you think this is needed by F1 23? Did you by any chance compare to the effects you get on windows? The thing is that most games actually don't use these effect types (`ff_friction`/`ff_inertia`/....). They often only use `ff_constant` to create all the effects (ACC for instance is such a game) and I think, rFactor only uses `ff_sine`. I'm not sure about F1 23, and I will check if it actually makes use of `ff_friction`.

> is this even the right tap or should i created it at pull requests?
Yes, this is the right tab. Pull requests are for integrating code from a fork. So, if you'd implemented it, and you would want to have this implementation pulled into my repo.

> i can close the other thread, no?
Which other thread? Generally, if you opened it and you're happy with the outcome, you can just close it (maybe add a comment why you believe closing was a good action).

**Comment by gotzl** [comment]
So, after some testing, I found that F1 23 uses `FF_DAMPER` and `FF_CONSTANT` only.
However, EA Sports WRC does use `FF_FRICTION`, and possibly Dirt Rally 2 as well. So, I've experimentally enabled it for further testing, but it doesn't feel quit alright just yet...

**Comment by dtran11** [comment]
I just tried out FF_FRICTION, it seems to feel just like FF_DAMPER to me.

**Comment by gotzl** [comment]
@dtran11 yep, the command send to the device is the same for FF_FRICTION / FF_INERTIA / FF_DAMPER, at least that's what I saw from WheelCheck. I thought, maybe games use different parameters to achieve the wished effect?
But yea, I'm a bit puzzled with this, too... I'm open for ideas ;)

**Comment by ciscoyo** [comment]
i´ll close this, coz with more knowledge i´m convinced by now, that all ff_eventcodes are present and they´re working as intended.
